common/protob/messages-bitcoin.proto

```diff
@@ -198,6 +198,18 @@ message SignTx {
     optional AmountUnit amount_unit = 11 [default=BITCOIN];    // show amounts in
     optional bool decred_staking_ticket = 12 [default=false];  // only for Decred, this is signing a ticket purchase
     optional bool serialize = 13 [default=true];               // serialize the full transaction, as opposed to only outputting the signatures
+    optional CoinJoinRequest coinjoin_request = 14;            // only for preauthorized CoinJoins
+
+    /**
+     * Signing request for a CoinJoin transaction.
+     */
+    message CoinJoinRequest {
+        required uint32 fee_rate = 1;                // coordination fee rate in units of 10^-8 percent
```

Comment:
This overflows `uint32` with 43 percent or more - not probable, but we might change it to `uint64` as protection.

Reply:
reasonable.
(fwiw the firmware currently still ignores uint sizes so this is not an issue; however, that is going to be fixed at some point)

Reply:
Firstly, a 43 % fee is crazy huge. However, I understand that we might want to represent percentages in other protobuf messages in the future and it would be nice to use a consistent format. The present scheme already allows 8 decimal digits of a percent, which seems way more precise than we might ever need. So rather than doubling the size of the data type, which would result in a maximum value of 184 billion %, I'd opt for decreasing the precision. If we go with 10<sup>–7</sup> or 10<sup>–6</sup>, then that can fit 429% or 4295% respectively and the precision is still great. We will probably never need more than 4 digits of percent precision anyway.

BTW note that the fee rate is specified in two places in protobuf `AuthorizeCoinJoin.max_coordinator_fee_rate` and `CoinJoinRequest.fee_rate`, so we definitely want to keep those consistent.

When CoinJoin was first implemented in July 2020, we had it at 10<sup>–9</sup>, see https://github.com/trezor/trezor-firmware/pull/1127. That means a maximum of 4.29 % could be represented. The meaning was also different (fee per anonymity unit vs. flat fee rate now). I recall that we discussed this at the standup back then and the argument was that 4 % (per anonymity unit) is huge.

In https://github.com/trezor/trezor-firmware/pull/2180 I changed the meaning to the flat fee rate and 8 digit precision to partially compensate the difference in meaning.

This is probably the last chance to easily change the precision, if we want to. It will also require changes in connect. After connect is deployed, it would probably be too complicated.

Reply:
+1 for 10^-6 precision. Let's not forget to change the `max_coordinator_fee_rate` too.

Reply:
@prusnak interesting point, but I think there is a difference. Here we are talking about precision of 10^-6 of a _percent_, whereas in LND I understand the precision to be 10^-6 of the whole. In other words if we want to follow LND, then we should be talking about 10^-4 of a percent here.

Reply:
ah, right, never mind

Reply:
So what will it be? 10^-4 (ppm, LND), 10^-6 (millionth of a percent), 10^-8 (current) etc.? To be honest, I have no strong preference. Anything between 4 and 8 is fine by me.

Reply:
10^-6 seems like a good middle ground.
Done in https://github.com/trezor/trezor-firmware/pull/2577/commits/52fca8cc50a4d8cb6f86959848d4f51c47e12ce0